src/components/position_calc/NLS_position_calc.py

```diff
@@ -0,0 +1,53 @@
+from components.position_calc import position_calc_utils
+import numpy as np
+from simulator_main import sim_config as cfg
+from sim_utils.common_types import *
+
+class NLS_position_calc:
+    
+    def __init__(self, name = "NLS_position_calc", optimization_type = OptimizationType.nelder_mead, 
```

Comment:
I'm hesitant to include parameters in the __init__(). Everything we need on initialization should be in the dictionary we pass in. 

The way all objects are initialized is via something like:
for param_dict in sim_config.component_dicts:
     // whatever Michael came up with

Since the body of the loop is generic for all objects, it's hard to pass in parameters differently for each object. Every object should only be initialized with the dictionary right?